protocol/bc/hash.go

```diff
@@ -111,12 +100,18 @@ func writeFastHash(w io.Writer, d []byte) error {
 	return err
 }
 
-// WriteTo writes p to w.
+// WriteTo satisfies the io.WriterTo interface.
 func (h *Hash) WriteTo(w io.Writer) (int64, error) {
-	n, err := w.Write(h[:])
-	return int64(n), err
+	return h.Byte32().WriteTo(w)
 }
 
-func (h *Hash) readFrom(r io.Reader) (int, error) {
-	return io.ReadFull(r, h[:])
+// WriteTo satisfies the io.ReaderFrom interface.
+func (h *Hash) ReadFrom(r io.Reader) (int64, error) {
+	var b32 Byte32
+	n, err := b32.ReadFrom(r)
```

Comment:
Simply not to repeat what `byte32` is already doing (and which I was supposed to reuse in `asset.go` but hadn't, and now have).